src/data/polynomial/ring_division.lean

```diff
@@ -87,10 +86,106 @@ begin
   rw [nat_degree_mul h2.1 h2.2], exact nat.le_add_right _ _
 end
 
+section roots
+
+open multiset
+
 local attribute [reducible] with_zero
 
-lemma exists_finset_roots : ∀ {p : polynomial R} (hp : p ≠ 0),
-  ∃ s : finset R, (s.card : with_bot ℕ) ≤ degree p ∧ ∀ x, x ∈ s ↔ is_root p x
+lemma degree_eq_zero_of_is_unit (h : is_unit p) : degree p = 0 :=
```

Comment:
Note to the reviewers: the lemmas before `root_multiplicity_zero` were moved from lower down in the file because the following lemmas depend on them. I didn't change anything about them apart from that. It might be better to put them in another file entirely, if you prefer.